zenoh/src/net/routing/interceptor/downsampling.rs

```diff
@@ -0,0 +1,160 @@
+//
+// Copyright (c) 2024 ZettaScale Technology
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+//
+// Contributors:
+//   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
+//
+
+//! ⚠️ WARNING ⚠️
+//!
+//! This module is intended for Zenoh's internal use.
+//!
+//! [Click here for Zenoh's documentation](../zenoh/index.html)
+
+use crate::net::routing::interceptor::*;
+use std::sync::{Arc, Mutex};
+use zenoh_config::DownsamplingItemConf;
+use zenoh_protocol::core::key_expr::OwnedKeyExpr;
+
+const RATELIMIT_STRATEGY: &str = "ratelimit";
```

Comment:
Strings are often used in the config, but as general approach during config parsing we generate the corresponding `enum` in such a way no `String` comparison needs to be done in the internal code.